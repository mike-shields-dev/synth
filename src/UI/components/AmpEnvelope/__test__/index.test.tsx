import { AmpEnvelope } from '..';
import { vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';

const validProps = {
    updateFocus: vi.fn(), 
    focus: 'ampEnvelope',
}

describe('AmpEnvelope', () => {
    it('matches snapshot', () => {
        const { container } = render(<AmpEnvelope {...validProps} />);

        expect(container).toMatchSnapshot();
    });

    it('gets focus when the user clicks on it', () => {
        render(<AmpEnvelope {...validProps} />);
        
        const ampEnvelopeEl = screen.getByText('Amp Envelope');

        fireEvent.click(ampEnvelopeEl);

        expect(validProps.updateFocus).toHaveBeenCalledWith('ampEnvelope');

        console.log(ampEnvelopeEl);
    });
});