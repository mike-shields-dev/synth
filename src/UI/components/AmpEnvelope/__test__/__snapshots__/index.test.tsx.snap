// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AmpEnvelope > matches snapshot 1`] = `
<div>
  <section
    aria-label="Amp Envelope"
    class="
        _SynthParameterGroup_541118
        _SynthParameterGroup--focus_541118"
    id="ampEnvelope"
  >
    <header>
      <h2>
        Amp Envelope
      </h2>
    </header>
    <div
      class="_SynthParameterGroup__controls_541118"
    >
      <div
        class="_Slider_8e0153"
      >
        <label
          for="envelope-A"
        >
          A
        </label>
        <input
          id="envelope-A"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="0"
        />
        <output
          aria-label="Envelope A"
          for="envelope-A"
        >
          0.0001
        </output>
      </div>
      <div
        class="_Slider_8e0153"
      >
        <label
          for="envelope-D"
        >
          D
        </label>
        <input
          id="envelope-D"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="6.348793487934879"
        />
        <output
          aria-label="Envelope D"
          for="envelope-D"
        >
          0.5
        </output>
      </div>
      <div
        class="_Slider_8e0153"
      >
        <label
          for="envelope-S"
        >
          S
        </label>
        <input
          id="envelope-S"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="127"
        />
        <output
          aria-label="Envelope S"
          for="envelope-S"
        >
          1
        </output>
      </div>
      <div
        class="_Slider_8e0153"
      >
        <label
          for="release-R"
        >
          R
        </label>
        <input
          id="release-R"
          max="127"
          min="0"
          step="0.05"
          type="range"
          value="12.698856988569887"
        />
        <output
          aria-label="Release R"
          for="release-R"
        >
          1
        </output>
      </div>
    </div>
  </section>
</div>
`;
